accepted/future-releases/primary-constructors/feature-specification.md

```diff
@@ -1004,6 +1040,15 @@ then _k2_ has an initializer list with the same elements in the same order.
 
 Finally, _k2_ is added to _D2_, and _D_ is replaced by _D2_.
 
+### Language versioning
+
+This feature is language versioned.
+
+*It introduces a breaking change in the grammar, which implies that
+developers must explicitly enable it. In particular, `factory() {}` in a
+class body used to be a method declaration. With this feature it will be
+a factory constructor declaration.*
```

Comment:
The feature also disallows `var x`, `final x` and `final T x` as parameters in all non-declaring constructors (Line 122/130.), which is much more far-reaching and potentially breaking.